# Day 11 Assignment 

Correaltion Analysis

In [63]:
import pandas as pd
import numpy as np

In [40]:
df = pd.read_csv("C:/Users/comp/Python Coding/LU AI-ML/Stats, Numpy, Pandas, Matplotlib/Day-7/Assignment/general_data.csv")

df.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [77]:
df.TotalWorkingYears.isna().sum()

9

In [65]:
df.TotalWorkingYears.mean()

11.279936378095888

In [85]:
df['TotalWorkingYears'] = df['TotalWorkingYears'].fillna(11.27)

In [89]:
df['NumCompaniesWorked'].mean()

2.6948303347756775

In [90]:
df['NumCompaniesWorked'] = df['NumCompaniesWorked'].fillna(2.69)

In [86]:
df['TotalWorkingYears'].isna().sum()

0

In [91]:
df['NumCompaniesWorked'].isna().sum()

0

In [44]:
from sklearn.preprocessing import LabelBinarizer

In [45]:
lb = LabelBinarizer()
df['Attrition'] = lb.fit_transform(df['Attrition'])

In [92]:
df.head()

Age  Attrition     BusinessTravel              Department  \
0   51          0      Travel_Rarely                   Sales   
1   31          1  Travel_Frequently  Research & Development   
2   32          0  Travel_Frequently  Research & Development   
3   38          0         Non-Travel  Research & Development   
4   32          0      Travel_Rarely  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                 6          2  Life Sciences              1           1   
1                10          1  Life Sciences              1           2   
2                17          4          Other              1           3   
3                 2          5  Life Sciences              1           4   
4                10          1        Medical              1           5   

   Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0  Female  ...                 1.0      Y                11              8   
1  Female  ...                 0.0      Y                23              8   
2    Male  ...                 1.0      Y                15              8   
3    Male  ...                 3.0      Y                11              8   
4    Male  ...                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [38]:
from scipy.stats import pearsonr

In [93]:
stats,p = pearsonr(df.Attrition, df.DistanceFromHome)

In [94]:
print(stats,p)

-0.009730141010179435 0.5182860428049617


Inference : There is slight negative correlation of -0.0097 between DistanceFromHome and Attrition.

In [49]:
stats, p = pearsonr(df.Attrition, df.Age)

In [50]:
print(stats,p)

-0.159205006865775 1.996801615893625e-26


Inference : The r value = -0.159, which shows a negative correlation between Age and Attrition.

In [51]:
stats, p = pearsonr(df.Attrition, df.Education)

In [52]:
print(stats,p)

-0.015111167710968753 0.3157293177118575


Inference : r value = -0.0151. Shows a negative correlation between Education and Attrition.

In [53]:
stats,p = pearsonr(df.Attrition, df.MonthlyIncome)

In [54]:
print(stats,p)

-0.031176281698114025 0.0384274849060192


Inference : r value = -0.031, which shows a negative correlation between MonthlyIncome and Attrition.

In [95]:
stats,p = pearsonr(df.Attrition, df.TotalWorkingYears)

In [96]:
print(stats,p)

-0.17011179875696666 5.4713069739411376e-30


Inference : r value = -0.170, which shows a negative correlation between TotalWorkingYears and Attrition.

In [97]:
stats, p = pearsonr(df.Attrition, df.YearsAtCompany)

In [98]:
print(stats,p)

-0.13439221398997386 3.163883122493571e-19


Inference : r value = -0.134. This indicates a negative correlation between YearsAtCompany and Attrition.

In [99]:
stats,p = pearsonr(df.Attrition, df.YearsWithCurrManager)

In [100]:
print(stats,p)

-0.1561993159016244 1.7339322652951965e-25


Inference : r value = -0.156. Shows a negative correlation between YearsWithCurrManager and Attrition.

In [101]:
stats,p = pearsonr(df.Attrition, df.PercentSalaryHike)

In [102]:
print(stats,p)

0.03253259489105223 0.030743386433369824


Inference : r value = 0.032. This shows a positive correlation between PercentSalaryHike and Attrition.

# Conclusion : 
    On most parameters at company XYZ, like DistanceFromHome, Age, Education, MonthlyIncome, TotalWorkingYears, YearsAtCompany, YearsWithCurrManager show a negative correlation whereas PercentSalaryHike shows a positive correlation.
    Considering the r value & p value of PercentSalaryHike, one may observe that this factor has a significant impact on Attrition rate of the company.